project-koku/koku
# [COST-2302] Cluster ID Attribute Error Alert

Fixes [COST-2302](https://issues.redhat.com/browse/COST-2302)

CONTEXT: When we switched to trino, we started running into an issue with cluster information not being in our `reporting_ocp_cluster` table, this table is filled during ocp trino processing. If a OCP source hasn't sent data in a long time (before the switch to trino), this table won't have info even though this cloud source used to connect to an OCP source as its infrastructure. If there is no info, this PR attempts to skip processing.

Changes proposed in this PR:
* If there is no cluster info for the OCP source, continue through the loop without attempting to process ocp on cloud to parquet
* If there is no cluster info for the OCP source, skip updating the ocp on cloud summary tables

Testing Instructions:
1. Set ENABLE_PARQUET_PROCESSING to false and spin up koku with trino around `make docker-up-min-trino`
2. Load some data with `make create-test-customer` followed by `make load-test-customer-data`
3. Let it all load, and hit up unleash `localhost:4242` and add the account (acct10001) to the cost-trino-process
4. create some more data and trigger a download for your aws source, (I use our `load_test_customer_data.sh` and the make command it goes with to do it, just comment out the nise report commands for everything but the ocp on aws aws source and comment out the trigger download call for everything but the aws source
5. If you are on main, you should see the AttributeError we have been seeing in prod, on this branch you should a nice new log message similar to: 
```
No cluster information available for OCP Provider: 48077900-5ec2-4046-8859-19506b8228ae, skipping OCP on Cloud summary table update for Azure source: 7e2ca510-2e49-4bbd-8f8b-eb8ea13c0b15.
```

---
Additional Context:
